pyqtgraph/graphicsItems/AxisItem.py

```diff
@@ -732,58 +749,49 @@ def tickSpacing(self, minVal, maxVal, size):
         dif = abs(maxVal - minVal)
         if dif == 0:
             return []
-
-        ## decide optimal minor tick spacing in pixels (this is just aesthetics)
-        optimalTickCount = max(2., log(size))
-
-        ## optimal minor tick spacing
-        optimalSpacing = dif / optimalTickCount
-
-        ## the largest power-of-10 spacing which is smaller than optimal
-        p10unit = 10 ** floor(log10(optimalSpacing))
-
-        ## Determine major/minor tick spacings which flank the optimal spacing.
-        intervals = np.array([1., 2., 10., 20., 100.]) * p10unit
-        minorIndex = 0
-        while intervals[minorIndex+1] <= optimalSpacing:
-            minorIndex += 1
-
+        
+        ref_size = 300. # axes longer than this display more than the minimum number of major ticks
+        minNumberOfIntervals = max(
+            2.25,       # 2.0 ensures two tick marks. Fudged increase to 2.25 allows room for tick labels. 
+            2.25 * self._tickDensity * sqrt(size/ref_size) # sub-linear growth of tick spacing with size
+        )
+
+        majorMaxSpacing = dif / minNumberOfIntervals        
+        p10unit = 10 ** floor(log10(majorMaxSpacing))  
+        for majorScaleFactor in (5.0, 2.0, 1.0):
+            if majorScaleFactor * p10unit <= majorMaxSpacing:
+                break # find the first value that is smaller or equal
+        majorInterval = majorScaleFactor * p10unit
+
+        minorMinSpacing = 2 * dif/size   # no more than one minor tick per two pixels
+        if majorScaleFactor == 1.0:
```

Comment:
I would recommend avoiding the equality comparison among floats, I know `1.0` is explicitely one of the values you set to, but I would feel better if you used `math.isclose(majorScaleFactor, 1.0)`

Reply:
I have been looking for a readable solution here. I think we can divide the power-of-ten unit by 10 so that all scale factors are integer. (0.5 --> 5;  1.0 --> 10 and so on)